### PR TITLE
dwarfs: 0.7.5 → 0.9.10

### PR DESCRIPTION
**Pull request by nh2**
The previous update forgot to update the hash, so the old source archive continued to be used:

https://github.com/NixOS/nixpkgs/pull/312656#discussion_r1650100776

## TODO

* [x] Fix build regarding `gtest`, getting the same error as:
  * https://github.com/mhx/dwarfs/issues/188
  * But using the submodule'd `gtest` doesn't seem to build either

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by tpwrules** [comment]
Can we also throw in the 0.9.10 upgrade and backport to 24.05?

**Comment by nh2** [comment]
> Can we also throw in the 0.9.10 upgrade

I'm looking into that, will probably push it on top of this PR.

> and backport to 24.05?

Probably, because it also has the bugged version.

File: pkgs/tools/filesystems/dwarfs/default.nix [hunk 1]
```diff
@@ -90,17 +91,31 @@ stdenv.mkDerivation {
   ];
 
   doCheck = true;
-  nativeCheckInputs = [ gtest ];
+  nativeCheckInputs = [
+    # https://github.com/mhx/dwarfs/issues/188#issuecomment-1907574427
+    # `dwarfs` sets C++20 as the minimum; thus the `gtest` headers, when included,
+    # refer to symbols that only exist in `.so` files compiled with that version.
+    (gtest.override { cxx_standard = "20"; })
```
Comment (tpwrules) [inline pkgs/tools/filesystems/dwarfs/default.nix:98]: This situation is kind of hokey, I guess it was too annoying to download gtest as a separate thing and move it into the build? That can be very particular with CMake.

**Review by tpwrules — COMMENTED** [review]
Overall looks good, a couple minor nits. Tested build and operation on x86_64-linux and aarch64-linux. Was not able to get cross to work unfortunately, probably issues with the CMake scripts. But I don't think that should hold this back.

File: pkgs/tools/filesystems/dwarfs/default.nix [hunk 2]
```diff
@@ -90,17 +91,31 @@ stdenv.mkDerivation {
...
+  ];
   # these fail inside of the sandbox due to missing access
   # to the FUSE device
-  GTEST_FILTER = "-dwarfs/tools_test.end_to_end/*:dwarfs/tools_test.mutating_ops/*";
+  GTEST_FILTER =
+    let
+      disabledTests = [
+        "dwarfs/tools_test.end_to_end/*"
+        "dwarfs/tools_test.mutating_ops/*"
```
[the remark this answers is not in the document]
Comment (nh2) [reply pkgs/tools/filesystems/dwarfs/default.nix]: Done.

File: pkgs/tools/filesystems/dwarfs/default.nix [hunk 3]
```diff
@@ -9,69 +9,70 @@
...
   ];
 
   nativeBuildInputs = [
     bison
     cmake
+    howard-hinnant-date # uses only the header-only parts
     pkg-config
+    range-v3 # header-only library
     ronn
+    (python3.withPackages (ps: [ ps.mistletoe ])) # fror man pages
```
[the remark this answers is not in the document]
Comment (nh2) [reply pkgs/tools/filesystems/dwarfs/default.nix]: Done.

File: pkgs/tools/filesystems/dwarfs/default.nix (hunk 1 above)
Comment (nh2) [reply pkgs/tools/filesystems/dwarfs/default.nix:100]: Yes, and this way at least when we patch `gtest` to work well in nixpkgs, we at least dont have to do it again in this package.

**Comment by nh2** [comment]
> Git push to origin failed for release-24.05 with exitcode 1

Not sure what this means, it created the backport PR successfully.

**Comment by SuperSandro2000** [comment]
the action does an extra git force push to trigger the actions on the PR

**Pull request merged**